nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -143,18 +155,35 @@ public Batch getBatch(String batchId) {
 
         // create the batch
         Batch batch = new Batch(batchId, parameters, getName(), getTransientStore());
-        AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest().withRoleSessionName(batch.getKey())
-                                                                     .withRoleArn(roleArn);
-        Credentials credentials = stsClient.assumeRole(assumeRoleRequest).getCredentials();
+
+        String bucketPath = concatPath(bucketPrefix, batchId) + "/";
+
+        String policy = replaceStringWithPlaceholders(policyTemplate, REGEX_BUCKET_PATH_PLACE_HOLDER, bucketPath) + "*";
+
+        GetFederationTokenRequest federationTokenRequest = new GetFederationTokenRequest()
+                .withPolicy(policy)
+                .withName(batchId + "_" + ClientLoginModule.getCurrentPrincipal().getOriginatingUser());
```

Comment:
It's worth extracting `batchId + "_" + ClientLoginModule.getCurrentPrincipal().getOriginatingUser()` into its own method for clarity, documentation, and maybe unit testing.